sdk/keyvault/azure-keyvault-keys/azure/keyvault/keys/crypto/_providers/local_provider.py

```diff
@@ -44,12 +45,12 @@ def supports(self, operation, algorithm):
 
     @property
     def key_id(self):
-        # type: () -> str
+        # type: () -> Optional[str]
         """The full identifier of the provider's key.
 
         :rtype: str
         """
-        return self._key.id
+        return self._key_id
```

Comment:
This will be either a valid Key Vault identifier or whatever the user gave us:
```suggestion
        return self._key.get("kid")
```
Does that remove the need to ask a caller to give us `_key_id` separately?

Reply:
My thinking here is that a KeyVaultKey provided to CryptographyClient could have a valid Key Vault identifier for its `id`, but that the JsonWebKey could have a different `kid`. I assumed it made more sense to return the Key Vault identifier in the operation result if possible, which is why I added the `_key_id` keyword argument